### PR TITLE
Made $ into a text escape character.

### LINKED ISSUE #1515: Allow parentheses in `[alt:]` text
As a side question: is there a way to have that numbering on top of the last note as in the source? `[alt:]` doesn't allow parentheses...

_Originally posted by @olivierberten in https://github.com/gregorio-project/gregorio/issues/1506#issuecomment-771885609_

### PR DESCRIPTION
**Pull request by henryso**
Fixes #1515

The lexer changes were pretty straightforward, but the gabc output was a bit tricky.  It seems to pass the tests satisfactorily, so I think it's working well enough though there may be some obscure bugs.

### CODE REVIEW

**Comment by henryso** [comment]
Note: the documentation doesn't build because of uprightpunctuminclinatumshift rename.  I think some merging needs to be done.

**Comment by rpspringuel** [comment]
I've been caught up in some problems in my own work, but I should be able to look at this later today.

**Comment by rpspringuel** [comment]
@henryso  I've also submitted a PR against this branch in your fork of the repository.

File: src/gabc/gabc-write.c [hunk 1]
```diff
@@ -238,6 +239,27 @@ static void gabc_print_char(FILE *f, const grewchar to_print)
...
+        case ':':
+        case '.':
+            if (write_state == GABC_IN_AUTO_PROTRUSION) {
+                break;
+            }
+            /* fall through */
+        case '-':
+        case '{':
+        case '}':
+        case '(':
```
[the remark this answers is not in the document]
Comment (henryso) [reply src/gabc/gabc-write.c:254]: There's no need.  Lexically, the close parentheses have no special meaning in normal text.

File: src/gabc/gabc-write.c [hunk 2]
```diff
@@ -238,6 +239,27 @@ static void gabc_print_char(FILE *f, const grewchar to_print)
...
+        case '[':
+        case ']':
+        case '<':
```
[the remark this answers is not in the document]
Comment (henryso) [reply src/gabc/gabc-write.c:257]: No, it doesn't. Lexically, the close angle bracket has no special meaning in normal text.

File: src/gabc/gabc-write.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (rpspringuel) [reply src/gabc/gabc-write.c:254]: But what about inside the notes?  Won't `[alt:4)]` create problems?

File: src/gabc/gabc-write.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (rpspringuel) [reply src/gabc/gabc-write.c:257]: Fair enough.

File: .gitignore [hunk 3]
```diff
@@ -126,3 +126,4 @@ gregorio-*
 ctan
 *.pyg
 .agignore
+*.orig
```
Comment (rpspringuel) [inline .gitignore:129]: Where is this extension from?  I don't recognize it.

File: src/gabc/gabc-write.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (henryso) [reply src/gabc/gabc-write.c:254]: I didn't think of that case.  Might have to tweak the lexer again.

[the remark this answers is not in the document]
Comment (henryso) [reply src/gabc/gabc-write.c:254]: I looked further into this.  At that point, the lexer is looking for a closing square bracket, so only the closing square bracket needs to be escaped, and that case is handled.  No need to escape the closing parenthesis.

File: .gitignore (hunk 3 above)
Comment (henryso) [reply .gitignore:129]: I use a tool for merging which leaves orig files.  I'd rather not commit them, and nothing else uses that, so I added it.

**Comment by henryso** [comment]
I don't think there are any remaining open questions here (but it is a bit hard to follow).

File: src/gabc/gabc-write.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (rpspringuel) [reply src/gabc/gabc-write.c:254]: Good to know.  Might it be worth having a test case which shows this happening as expected?

File: .gitignore (hunk 3 above)
Comment (rpspringuel) [reply .gitignore:129]: No problem then.

File: src/gabc/gabc-write.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (henryso) [reply src/gabc/gabc-write.c:254]: Good idea.  I'll add it.

**Comment by henryso** [comment]
Looks like the gabc writer doesn't escape the bracketed instructions inside the notes.  I'll working on that.  Might take a bit.

**Comment by henryso** [comment]
OK, how does this look?

**Comment by rpspringuel** [comment]
I think that about covers it.
